awx/main/models/rbac.py

```
@@ -242,270 +177,29 @@ def description(self):
-                )
-            )
-            .distinct()
-            .values_list('ancestor', flat=True)
-        )
+        if user.is_superuser or user.is_system_auditor:
+            return roles_qs
+
+        from ansible_base.rbac.models import RoleEvaluation
+
+        q = RoleEvaluation.objects.filter(role__in=user.has_roles.all()).values_list('object_id', 'content_type_id').query
+        return roles_qs.extra(where=[f'(object_id,content_type_id) in ({q})'])
```

Ruff (0.15.10)

[error] Use of Django `extra` can lead to SQL injection vulnerabilities

(S610)